### PR TITLE
Improve defines for I2C and I2S features

### PR DESCRIPTION
**Pull request by xluciano**

### CODE REVIEW

File: doc/programming_guide/ffd/deploying/configuration.rst [hunk 1]
```diff
@@ -34,17 +34,17 @@ If options are changed, the application firmware must be rebuilt.
...
+     - Enables/disables sending the intent message over |I2C| master
      - 1
-   * - appconfINTENT_I2C_OUTPUT_DEVICE_ADDR
+   * - appconfINTENT_I2C_MASTER_DEVICE_ADDR
      - Sets the address of the |I2C| device receiving the intent via the |I2C| master interface
      - 0x01
-   * - appconfI2C_SLAVE_ENABLED
-     - Enables/disables the |I2C| slave mode to read the device register with the intent message
+   * - appconfINTENT_I2C_SLAVE_POLLED_ENABLED
+     - Enables/disables allowing polling the intent message via |I2C| slave
```
Comment (mbanth) [inline doc/programming_guide/ffd/deploying/configuration.rst]: I suggest changing to `Enables/disables allowing another device to poll the intent message via |I2C| slave`

File: doc/programming_guide/ffd/deploying/configuration.rst [hunk 2]
```diff
@@ -102,31 +102,30 @@ The |I2C| master is used when the FFD example asynchronously sends intent messag
...
-  - ``appconfINTENT_I2C_OUTPUT_ENABLED`` to 0, this will disable the |I2C| master interface.
+  - ``appconfINTENT_I2C_MASTER_OUTPUT_ENABLED`` to 0, this will disable the |I2C| master interface after initialization.
 
 The handling of the |I2C| slave registers is done in the ``examples\ffd\src\i2c_reg_handling.c`` file. The variable ``appconfINTENT_I2C_REG_ADDRESS`` is used in the callback function ``read_device_reg()``.
 
 Configuring the |I2S| interface
 -------------------------------
 
-The |I2S| interface is used to receive the audio data from the host. The |I2S| interface can be configured as either a master or a slave.
+The |I2S| interface is used to send the intent audio to the DAC, and/or to receive the audio samples from the host. The |I2S| interface can be configured as either a master or a slave.
```
Comment (mbanth) [inline doc/programming_guide/ffd/deploying/configuration.rst]: I don't clearly understand the meaning of `intent audio`.

**Review by mbanth — CHANGES_REQUESTED** [review]

File: doc/programming_guide/ffd/deploying/configuration.rst (hunk 1 above)
Comment (xluciano) [reply doc/programming_guide/ffd/deploying/configuration.rst]: Fixed in https://github.com/lucianomartin/sln_voice/pull/1/commits/e30e4603aca7a98dcbc5ec7757ef893a8840ee83.

File: doc/programming_guide/ffd/deploying/configuration.rst (hunk 2 above)
Comment (xluciano) [reply doc/programming_guide/ffd/deploying/configuration.rst]: I added more details in https://github.com/lucianomartin/sln_voice/pull/1/commits/e30e4603aca7a98dcbc5ec7757ef893a8840ee83.

I didn't manage to test the audio files, so I am not sure how they work. Let me know if I should create a story to understand them better.

File: doc/programming_guide/ffd/overview.rst [hunk 3]
```diff
@@ -13,6 +13,13 @@ The examples using the microphone array as the audio source include an audio pip
     #. Noise Suppressor (NS)
     #. Adaptive Gain Control (AGC)
 
+The FFD examples provide several options to inform the host of a possible intent detected by the intent engine. The device can notify the host by:
+
+  - sending the intent ID over UART interface upon the detection
```
Comment (mbanth) [inline doc/programming_guide/ffd/overview.rst]: `over UART` -> `over a UART`

Also a similar change on line 19.

Comment (mbanth) [inline doc/programming_guide/ffd/overview.rst]: `upon the detection` -> `upon detecting the intent`

Also, the same change on line 19.

File: doc/programming_guide/ffd/overview.rst [hunk 4]
```diff
@@ -13,6 +13,13 @@ The examples using the microphone array as the audio source include an audio pip
...
+  - sending the intent ID over |I2C| master interface upon the detection
+  - allowing the host polling the last detected intent ID over |I2C| slave interface
```
Comment (mbanth) [inline doc/programming_guide/ffd/overview.rst]: `the host polling` -> `the host to poll`

Comment (mbanth) [inline doc/programming_guide/ffd/overview.rst]: `over |I2C| interface` -> `over the |I2C| interface`

**Review by mbanth — CHANGES_REQUESTED** [review]

File: doc/programming_guide/ffd/overview.rst (hunk 3 above)
Comment (xluciano) [reply doc/programming_guide/ffd/overview.rst]: Fixed in https://github.com/lucianomartin/sln_voice/pull/1/commits/2e4c50e53433b45b7e70b0990856266d7f306f5a.

File: doc/programming_guide/ffd/overview.rst (hunk 4 above)
Comment (xluciano) [reply doc/programming_guide/ffd/overview.rst]: Fixed in https://github.com/lucianomartin/sln_voice/pull/1/commits/2e4c50e53433b45b7e70b0990856266d7f306f5a.

Comment (xluciano) [reply doc/programming_guide/ffd/overview.rst]: Fixed in https://github.com/lucianomartin/sln_voice/pull/1/commits/2e4c50e53433b45b7e70b0990856266d7f306f5a.

File: doc/programming_guide/ffd/overview.rst (hunk 3 above)
Comment (xluciano) [reply doc/programming_guide/ffd/overview.rst]: Fixed in https://github.com/lucianomartin/sln_voice/pull/1/commits/2e4c50e53433b45b7e70b0990856266d7f306f5a.

**Review by mbanth — APPROVED** [review]

**Pull request merged**